src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let copy = { ...state };
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        copy = Object.assign({}, copy, object.extraData);
```

Comment:
The use of `Object.assign` here is correct for adding properties, but ensure that `copy` is a new object each time. This line is fine as it creates a new object.